src/idl/src/directive.c

```diff
@@ -147,39 +147,32 @@ static idl_retcode_t push_line(idl_pstate_t *pstate, struct line *dir)
       }
     }
 
-    if ((ret = idl_normalize_path(dir->path, &norm)) < 0)
+    if ((ret = idl_normalize_path(dir->path, &norm)) < 0) {
+      idl_error(pstate, NULL, "Invalid line marker: path '%s' not found", dir->path);
       return ret;
+    }
     free(dir->path);
     dir->path = norm;
+    assert(dir->file);
 
-    if (dir->flags & RETURN_TO_FILE) {
-      for (; src; src = src->parent)
-        if (src->path->name && strcmp(src->path->name, dir->path) == 0)
-          break;
-      if (src) {
-        pstate->scanner.position.source = src;
-        pstate->scanner.position.file = src->path;
-      } else {
-        idl_error(pstate, idl_location(dir),
-          "Invalid line marker, file '%s' not on include stack", dir->path);
-        return IDL_RETCODE_SEMANTIC_ERROR;
-      }
+    /* reuse normalized filename if include is absolute */
+    if (idl_isabsolute(dir->file)) {
+      free(dir->file);
+      if (!(dir->file = idl_strdup(dir->path)))
+        return IDL_RETCODE_NO_MEMORY;
+    /* use original filename by default */
```

Comment:
I’m looking at the diff, so I may be wrong, but the indentation looks off to me here

Reply:
The indentation was not really off, it applies to the `else` block and matches indentation of that line. But I agree that it looks better when moved to inside the block, so I've fixed that in [e652aeb](https://github.com/eclipse-cyclonedds/cyclonedds/pull/1324/commits/e652aeb95f5a4bfde8923b89e592aaa2f77f91fc).